### PR TITLE
solution

### PR DESCRIPTION
**Pull request by vlad-shulzhenko**

### CODE REVIEW

**Review by io-med — APPROVED** [review]
Good job!

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,7 +1,71 @@
...
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+type SortOrder = 'asc' | 'desc';
+
+function getAverage(student: Student): number {
```
Comment (aliceblunt8) [inline src/sortStudents.ts:22]:
```suggestion
function getAverageGrade(student: Student): number {
```

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,7 +1,71 @@
...
+  /* eslint-disable max-len */
+  const average: number = student.grades.reduce((total: number, grade: number) => (
+    total + grade)) / student.grades.length;
+
+  return average;
+}
```
Comment (aliceblunt8) [inline src/sortStudents.ts:28]:
```suggestion
  const grades: number = student.grades
    .reduce((total: number, grade: number) => total + grade);

  return grades / student.grades.length;
}
```

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,7 +1,71 @@
...
+
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  const studentsCopy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return order === 'asc'
+        ? studentsCopy.sort((first: Student, second: Student) => (
+          first.name.localeCompare(second.name)))
```
Comment (aliceblunt8) [inline src/sortStudents.ts:37]: use sortBy everywhere instead object key names

```suggestion
          first[sortBy]localeCompare(second[sortBy])))
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**